### PR TITLE
EN-6612 added ports range capability on network messenger

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added port range capability on the network messenger

### CODE REVIEW

File: p2p/errors.go [hunk 1]
```diff
@@ -67,23 +67,29 @@ var ErrTooManyGoroutines = errors.New(" number of goroutines exceeded")
...
 
-// ErrWatchdogAlreadyStarted signals that a peer discovery watchdog is already started
-var ErrWatchdogAlreadyStarted = errors.New("peer discovery watchdog is already started")
+// ErrInvalidEndingPortValue signals that an invalid ending port value has been provided
+var ErrInvalidEndingPortValue = errors.New("invalid ending port value")
 
-// ErrWatchdogNotStarted signals that a peer discovery watchdog is not started
-var ErrWatchdogNotStarted = errors.New("peer discovery watchdog is not started")
+// ErrEndPortSmallerThanStartPort signals that the ending port value is smaller than the starting port value
+var ErrEndPortSmallerThanStartPort = errors.New("ending port value is less than the starting port value")
```
Comment (SebastianMarian) [inline p2p/errors.go:83]: is smaller

File: p2p/libp2p/ports.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+		return val, nil
+	}
+
+	ports := strings.Split(port, "-")
+	if len(ports) != 2 {
+		return 0, fmt.Errorf("%w, provided port string `%s` is not in the correct format, expected `start-end`", p2p.ErrInvalidPortsRangeString, port)
+	}
+
+	startPort, err := strconv.Atoi(ports[0])
```
Comment (SebastianMarian) [inline p2p/libp2p/ports.go:30]: This does not should be checked for negative value?

File: p2p/libp2p/ports.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	if err != nil {
+		return 0, p2p.ErrInvalidStartingPortValue
+	}
+
+	endPort, err := strconv.Atoi(ports[1])
```
Comment (SebastianMarian) [inline p2p/libp2p/ports.go:35]: This does not should be checked for negative value?

File: p2p/errors.go (hunk 1 above)
Comment (iulianpascalau) [reply p2p/errors.go:83]: done

File: p2p/libp2p/ports.go (hunk 2 above)
Comment (iulianpascalau) [reply p2p/libp2p/ports.go:30]: it is checked against minRangePortValue which is 1025.

File: p2p/libp2p/ports.go (hunk 3 above)
Comment (iulianpascalau) [reply p2p/libp2p/ports.go:35]: it is checked against startPort which should be greater than minRangePortValue

File: p2p/libp2p/ports.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+			continue
+		}
+
+		return p, nil
+	}
+
+	return 0, fmt.Errorf("%w, range %d-%d", p2p.ErrNoFreePortInRange, startPort, endPort)
+}
+
+func testFreePort(port int) error {
```
Comment (sasurobert) [inline p2p/libp2p/ports.go:75]: rename to checkFreePort

File: p2p/libp2p/ports.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/core/random"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+type testPortHandler = func(int) error
```
Comment (sasurobert) [inline p2p/libp2p/ports.go:13]: this type def is not necessary. if yes rename.

Comment (iulianpascalau) [reply p2p/libp2p/ports.go:13]: removed type definition. However, I kept the function pointer in order to better test the functionality.

File: p2p/libp2p/ports.go (hunk 4 above)
Comment (iulianpascalau) [reply p2p/libp2p/ports.go]: done

**Pull request merged**